### PR TITLE
compiler: remove some unused routines and variables

### PR DESCRIPTION
**Pull request by zerbina**
## Summary
Remove unused routines and variables from modules related to the VM and code generation. They were either never used or became unused as a side-effect of other removals / cleanup.

### CODE REVIEW

File: compiler/backend/ccgliterals.nim [hunk 1]
```diff
@@ -27,9 +27,6 @@ template detectVersion(field, corename) =
 proc detectStrVersion(m: BModule): int =
   detectVersion(strVersion, "nimStrVersion")
 
-proc detectSeqVersion(m: BModule): int =
-  detectVersion(seqVersion, "nimSeqVersion")
-
 # ----- Version 1: GC'ed strings and seqs --------------------------------
 
 proc genStringLiteralDataOnlyV1(m: BModule, s: string): Rope =
```
Comment (saem) [inline compiler/backend/ccgliterals.nim:32]: On the phone and doing sketchy GitHub searches but this proc might be unused now.

File: compiler/backend/ccgliterals.nim [hunk 2]
```diff
@@ -92,17 +89,6 @@ proc genStringLiteralV2Const(m: BModule; n: PNode; isConst: bool): Rope =
 
 # ------ Version selector ---------------------------------------------------
 
-proc genStringLiteralDataOnly(m: BModule; s: string; info: TLineInfo;
-                              isConst: bool): Rope =
-  case detectStrVersion(m)
-  of 0, 1: result = genStringLiteralDataOnlyV1(m, s)
-  of 2:
-    result = getTempName(m)
-    genStringLiteralDataOnlyV2(m, s, result, isConst)
```
Comment (saem) [inline compiler/backend/ccgliterals.nim:101]: Maybe thing one is unused, too

**Review by saem — APPROVED** [review]
There is a slim chance I might have one or two more dead procs, hopefully they're not false positives.

Thanks, the less code the better.

File: compiler/backend/ccgliterals.nim (hunk 1 above)
Comment (zerbina) [reply compiler/backend/ccgliterals.nim:32]: Not yet, `genStringLiteralV1` still uses it. Removing `refc` and friend will indeed make it obsolete, however.

File: compiler/backend/ccgliterals.nim (hunk 2 above)
Comment (zerbina) [reply compiler/backend/ccgliterals.nim:101]: It's used by `genStringLiteralV2`.

**Comment by zerbina** [comment]
Thanks for taking a look, @saem.

bors r+

**Pull request merged**